### DISCUSSION TITLE
Fetching and Globally Accessing Initial Data in Next.js for Both Client and Server Sides

### DISCUSSION THREAD

**Discussion by frezzzen** [vercel/next.js#52552] [Help]
### Summary

Hello,

I'm currently developing an application that requires two critical elements upon initial load: translations and a configuration object. These elements are not exclusively for client-side usage, but are also crucial on the server-side.

Here's the challenge I'm facing: Before I can retrieve specific data for a given page, I need to access a configuration object. This object contains various parameters that I need to include in my requests to the server.

I've considered fetching this data in the RootLayout and then making it available across all pages of the website. This was relatively easy to implement in Nuxt, but I've been having trouble finding an analogous solution in Next.js.

So, my question is: How can I fetch this data on the initial load and make it accessible throughout my application, both on the client-side and server-side, using Next.js?

**Comment by icyJoseph** [comment]
I think, if the configuration and translations are loaded by fetch, without dependency on the current path, then the answer is to just fetch... because it'll be deduped, and cached. 

So if you fetch in the root layout, other fetch requests made down the React tree see the same data.

- https://nextjs.org/docs/app/building-your-application/data-fetching#automatic-fetch-request-deduping

And then, for client components, in the root layout you could pass this to a Provider, and have the clients consume it via context, right?

If you don't want to rely on fetch cache, you could store the data on a Map or something, as a singleton and read it downstream in your app too.
